packages/alfa-rules/src/sia-r8/rule.ts

```diff
@@ -28,34 +28,51 @@ export default Rule.Atomic.of<Page, Element>({
         return getElementDescendants(document, Node.fullTree).filter(
           and(
             hasNamespace(Namespace.HTML),
-            hasRole(
-              device,
-              "checkbox",
-              "combobox",
-              "listbox",
-              "menuitemcheckbox",
-              "menuitemradio",
-              "radio",
-              "searchbox",
-              "slider",
-              "spinbutton",
-              "switch",
-              "textbox",
+            or(
+              hasRole(
+                device,
+                "checkbox",
+                "combobox",
+                "listbox",
+                "menuitemcheckbox",
+                "menuitemradio",
+                "radio",
+                "searchbox",
+                "slider",
+                "spinbutton",
+                "switch",
+                "textbox",
+              ), 
+              and(hasName("input"), or(hasInputType("password"), hasInputType("color"), 
+                hasInputType("date"), hasInputType("datetime-local"), hasInputType("file"), 
+                hasInputType("month"), hasInputType("time"), hasInputType("week"))),
```

Comment:
```suggestion
              hasInputType("password", "color", "date", "datetime-local", "file", "month", "time", "week"),
```
`hasInputType already checks for name, and accepts a list of allowed types.